### PR TITLE
Add support for generating a Calico CNI Spec

### PR DESCRIPTION
**Pull request by varunmar**
This is only to generate the spec for calico, using the same spec name. Changes to the calico daemonset to use this file instead of generating its own are upcoming.

### CODE REVIEW

File: netd.yaml [hunk 1]
```diff
@@ -91,6 +91,43 @@ data:
         }
       ]
     }
+  cni_calico_spec_template: |-
```
Comment (jingax10) [inline netd.yaml]: nit: s/cni_calico_spec_template/calico_cni_spec_template/

File: netd.yaml [hunk 2]
```diff
@@ -132,6 +169,11 @@ spec:
               configMapKeyRef:
                 name: netd-config
                 key: cni_spec_template
+          - name: CNI_CALICO_SPEC_TEMPLATE
```
Comment (jingax10) [inline netd.yaml]: nit: s/CNI_CALICO_SPEC_TEMPLATE/CALICO_CNI_SPEC_TEMPLATE/

File: scripts/install-cni.sh [hunk 3]
```diff
@@ -16,8 +16,10 @@
 
 set -u -e
 if [ "${ENABLE_CALICO_NETWORK_POLICY}" == "true" ]; then
-  echo "Calico Network Policy is enabled by ENABLE_CALICO_NETWORK_POLICY. Disabling CNI Spec generation."
-  exit 0
+  echo "Calico Network Policy is enabled by ENABLE_CALICO_NETWORK_POLICY. Generating Calico spec."
```
Comment (jingax10) [inline scripts/install-cni.sh]: nit: s/Generating Calico spec/Generating Calico CNI spec/

File: netd.yaml (hunk 1 above)
Comment (varunmar) [reply netd.yaml]: Done.

File: netd.yaml (hunk 2 above)
Comment (varunmar) [reply netd.yaml]: Done.

File: scripts/install-cni.sh (hunk 3 above)
Comment (varunmar) [reply scripts/install-cni.sh]: Done.

**Comment by varunmar** [comment]
Tested with manual fixes to calico-node-daemonset.yaml
The CNI spec template works, so I'm going to merge this in.

**Comment by jingax10** [comment]
/lgtm

@varunmar, thanks for verification! Please also port your change to custom calico yaml change for GKE.

**Pull request merged**